code/crates/sync/src/handle.rs

```diff
@@ -515,6 +545,33 @@ where
     Ok(())
 }
 
+/// Request values for this specific range from a peer.
+/// Should only be used when re-requesting a partial range of values from a peer.
+async fn request_values_range<Ctx>(
+    co: Co<Ctx>,
+    state: &mut State<Ctx>,
+    metrics: &Metrics,
+    range: RangeInclusive<Ctx::Height>,
+) -> Result<(), Error<Ctx>>
+where
+    Ctx: Context,
+{
+    // NOTE: We do not perform a `max_parallel_requests` check here in contrast to what is done, for
+    // example in `request_values`. This is because `request_values_range` is only called for retrieving
+    // partial responses, which means the original request is not on the wire anymore.
```

Comment:
I understand the original request is not in the wire (that is, its response was received) but it's still in `state.pending_requests`. At least for the received range that is being processed. Is this correct? Then, after the new request issued here, there will be `max_parallel_requests + 1` active requests in `state.pending_requests`. I'm not sure if this is a problem, but two requests will need to be removed from `state.pending_requests` before a new one can be issued.